src/uu/ls/src/dired.rs

```diff
@@ -0,0 +1,196 @@
+// This file is part of the uutils coreutils package.
+//
+// For the full copyright and license information, please view the LICENSE
+// file that was distributed with this source code.
+// spell-checker:ignore dired subdired dired
+
+use crate::Config;
+use std::fmt;
+use std::io::{BufWriter, Stdout, Write};
+use uucore::error::UResult;
+
+#[derive(Debug, Clone)]
+pub struct BytePosition {
+    pub start: usize,
+    pub end: usize,
+    pub is_total: Option<bool>,
```

Comment:
Would it be possible to model the total some other way? Maybe it could be a boolean on `DiredOutput` called `just_printed_total`?